zerver/views/message_summary.py

```diff
@@ -0,0 +1,204 @@
+import json
+from typing import Any
+
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+from django.utils.timezone import now as timezone_now
+from django.utils.translation import gettext as _
+from pydantic import Json
+
+from analytics.lib.counts import COUNT_STATS, do_increment_logging_stat
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.message import messages_for_ids
+from zerver.lib.narrow import (
+    LARGER_THAN_MAX_MESSAGE_ID,
+    NarrowParameter,
+    clean_narrow_for_message_fetch,
+    fetch_messages,
+)
+from zerver.lib.response import json_success
+from zerver.lib.typed_endpoint import typed_endpoint
+from zerver.models import UserProfile
+
+# Maximum number of messages that can be summarized in a single request.
+MAX_MESSAGES_SUMMARIZED = 100
+# Price per token for input and output tokens.
+# These values are based on the pricing of the Bedrock API
+# for Llama 3.3 Instruct (70B).
+# https://aws.amazon.com/bedrock/pricing/
+# Unit: USD per 1 billion tokens.
+#
+# These values likely will want to be declared in configuration,
+# rather than here in the code.
+OUTPUT_COST_PER_GIGATOKEN = 720
+INPUT_COST_PER_GIGATOKEN = 720
+
+
+def format_zulip_messages_for_model(zulip_messages: list[dict[str, Any]]) -> str:
+    # Format the Zulip messages for processing by the model.
+    #
+    # - We don't need to encode the recipient, since that's the same for
+    #   every message in the conversation.
+    # - We use full names to reference senders, since we want the
+    #   model to refer to users by name. We may want to experiment
+    #   with using silent-mention syntax for users if we move to
+    #   Markdown-rendering what the model returns.
+    # - We don't include timestamps, since experiments with current models
+    #   suggest they do not make relevant use of them.
+    # - We haven't figured out a useful way to include reaction metadata (either
+    #   the emoji themselves or just the counter).
+    # - Polls/TODO widgets are currently sent to the model as empty messages,
+    #   since this logic doesn't inspect SubMessage objects.
+    zulip_messages_list = [
+        {"sender": message["sender_full_name"], "content": message["content"]}
+        for message in zulip_messages
+    ]
+    return json.dumps(zulip_messages_list)
```

Comment:
This should use `orjson`

Reply:
Okay, will add that change as a commit in #32895.

Reply:
oh! I see.